driver-core/src/test/resources/auth/connection-string.json

```diff
@@ -3,101 +3,91 @@
     {
       "description": "should use the default source and mechanism",
       "uri": "mongodb://user:password@localhost",
-      "hosts": null,
       "valid": true,
-      "warning": false,
-      "auth": {
+      "credential": {
         "username": "user",
         "password": "password",
-        "db": "admin"
-      },
-      "options": null
+        "source": "admin",
+        "mechanism": null,
+        "mechanism_properties": null
+      }
     },
     {
       "description": "should use the database when no authSource is specified",
       "uri": "mongodb://user:password@localhost/foo",
-      "hosts": null,
       "valid": true,
-      "warning": false,
-      "auth": {
+      "credential": {
         "username": "user",
         "password": "password",
-        "db": "foo"
-      },
-      "options": null
+        "source": "foo",
+        "mechanism": null,
+        "mechanism_properties": null
+      }
     },
     {
       "description": "should use the authSource when specified",
       "uri": "mongodb://user:password@localhost/foo?authSource=bar",
-      "hosts": null,
       "valid": true,
-      "warning": false,
-      "auth": {
+      "credential": {
         "username": "user",
         "password": "password",
-        "db": "bar"
-      },
-      "options": null
+        "source": "bar",
+        "mechanism": null,
+        "mechanism_properties": null
+      }
     },
     {
       "description": "should recognise the mechanism (GSSAPI)",
       "uri": "mongodb://user%40DOMAIN.COM@localhost/?authMechanism=GSSAPI",
-      "hosts": null,
       "valid": true,
-      "warning": false,
-      "auth": {
+      "credential": {
         "username": "user@DOMAIN.COM",
         "password": null,
-        "db": "$external"
-      },
-      "options": {
-        "authmechanism": "GSSAPI"
+        "source": "$external",
+        "mechanism": "GSSAPI",
+        "mechanism_properties": {
+          "SERVICE_NAME": "mongodb"
+        }
       }
     },
     {
       "description": "should ignore the database (GSSAPI)",
       "uri": "mongodb://user%40DOMAIN.COM@localhost/foo?authMechanism=GSSAPI",
-      "hosts": null,
       "valid": true,
-      "warning": false,
-      "auth": {
+      "credential": {
         "username": "user@DOMAIN.COM",
         "password": null,
-        "db": "$external"
-      },
-      "options": {
-        "authmechanism": "GSSAPI"
+        "source": "$external",
+        "mechanism": "GSSAPI",
+        "mechanism_properties": {
+          "SERVICE_NAME": "mongodb"
```

Comment:
Is this the only place where we include a default value for a mechanism property?  I don't recall what moved us to add this, but it does require a special case in the test, since at least in the Java driver default mechanism property values are applied in the authenticator not the `MongoCredential`.

Reply:
I counted 4 tests in `connection-string.json` where the default value for the mechanism property `SERVICE_NAME` was used.

Reply:
Right.  I'm now wondering whether this is going to cause problems for drivers because we're assuming that the credential object is where the default is applied.

But I guess we'll see if others complain.